kythe/go/extractors/config/runextractor/mavencmd/pom_xml_modifier.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mavencmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/beevik/etree"
+)
+
+// PreProcessPomXML takes a pom.xml file and either verifies that it already has
+// the bits necessary to specify a separate compiler on commandline, or adds
+// functionality by dropping in a maven-compiler-plugin to the build.
+//
+// Note this potentially modifies the input file, so make a copy beforehand if
```

Comment:
s/potentially modifies/overwrites/

Reply:
Relatedly: Please document clearly that the file may be modified even if the function reports an error, so that the caller is responsible for cleanup either way.

Reply:
I worry this might be somewhat brittle to use. It might be worthwhile to have this function create a backup and return it to the caller, so they can choose to clean up and/or restore.

```
// AddPluginToPOM injects …
// A backup is made of the original file before any changes are made. The caller
// is responsible for releasing or restoring the backup.
func AddPluginToPOM(pomFile string) (*backup.Backup, error) {
    bkup, err := backup.New(pomFile)
    if err != nil {
        return nil, err
    }
    // do real work
    return bkup, nil
}
```

(To make this nice, we'd probably want to make the backup methods no-op on a nil receiver)

Again not necessarily something for this change, though.

Reply:
Hmm I'd kinda leave it as-is for now and keep backup done one level up.  That way if we either need multiple modifications we can do it in one spot where we really care, and if we end up not needing back up at all it removes pretty cleanly too.